CONTRIBUTING.md

```diff
@@ -65,6 +65,15 @@ cargo install rusty-hook
 rusty-hook init
 ```
 
+If you are updating docs, you will also need to install
+[github.com/executablebooks/mdformat](https://github.com/executablebooks/mdformat):
+
+```bash
+python3 -m venv .venv
+source .venv/bin/activate
+python3 -m pip install mdformat
+```
```

Comment:
Alternatively you can run `./tools/devtool checkstyle`

Reply:
ah so I suggested this because of

https://github.com/firecracker-microvm/firecracker/blob/45a1a1be3efde06f06034fbe1798c3747c1424cd/pre-commit#L37-L39
(the pre-commit explicitly calls to `mdformat`.)

Now that I look at it, the pre-commit also has these other dependencoes explicitly called out

https://github.com/firecracker-microvm/firecracker/blob/45a1a1be3efde06f06034fbe1798c3747c1424cd/pre-commit#L32-L36


And for completeness, this is what I get when running `./tools/devtool checkstyle` (i spin up EC2s for actually running the code, as recommended in the getting started guide, but otherwise keep my fork on a mac):
```
/Users/user/go/src/github.com/firecracker/tools/functions: line 11: BASH_SOURCE: bad array subscript
[Firecracker ] /dev/kvm not found. Aborting.
```

So I think this PR in short is more of a question of: would it be ok to have a list of all depencies listed along with instructions for how to install them? or should the pre-commit and other such bits rely on `./tools/devtool` (and explicitly state that people should do code commiting on a linux machine that is configured in a manner similar to the machine we use for tests)? or some other option I'm not thinking of at the moment?

wdyt?

Reply:
@pb8o should I make an issue to use `devtool checkstyle` in the pre-commit ?

Reply:
The difference between `checkstyle` and the precommit hook is that the precommit hook in its current form will fix the formatting issues for you, while `checkstyle` is only going to yell at you :(

Nevertheless, having the style thingy fail on the kvm checks is still a bit silly, maybe we can have an issue to fix that instead? :o

Reply:
Actually, after leaving my other comment, I think its probably better to have a `./tools/devtool fixstyle` command that runs all the style-fixing stuff that the pre-commit hook tries to do inside of docker (so that it picks up all the correct versions of the tools and such), and then the pre-commit hook can just point to that?

Reply:
Looks good! Thanks for doing that!

Reply:
Thank you @roypat and @pb8o for the reviews!
I rewrote to capture the latest round of ideas https://github.com/firecracker-microvm/firecracker/issues/4747 ptal

and also updated the PR to actually make use of the `./tools/devctr` poetry project - instead fo doing a weird rando thing with a python virtualenv.